netstat/netstat_linux.go

```diff
@@ -213,13 +213,13 @@ func (p *procFd) iterFdDir() {
 					return
 				}
 				n, err := stat.Read(buf[:])
-				stat.Close()
 				if err != nil {
 					return
 				}
 				z := bytes.SplitN(buf[:n], []byte(" "), 3)
 				name := getProcName(z[1])
 				p.p = &Process{p.pid, name}
+				stat.Close()
```

Comment:
What happens, do you think, if `stat.Read()` fails? You would end up with an open
file descriptor.